File: CodingSamples/Java/Language/InterfaceTest1/src/Account.java
package banking;

//a class defined with abstract modifier cannot be instantiated
public abstract class Account {

	long id;
	protected double balance;

	public long getId() {
		return id;
	}

	public double getBalance() {
		return balance;
	}

	//an instance method declared with abstract modifier
	//is a pure method and it must be overridden by the subclass
	public abstract void deposit(double amount);

	public abstract void withdraw(double amount) throws InsufficientFundsException;

	//a method declared with final modifier cannot be overridden in a subclass,
	//for such a method (slow) dynamic binding is not required at runtime
	public final void transfer(double amount, Account that) throws InsufficientFundsException {
		if(this == that)
			throw new IllegalTransferException();
		this.withdraw(amount);
		that.deposit(amount);
	}
}

File: CodingSamples/DotNet/Language/InterfaceTest1/BankLib/Account.cs
namespace Banking;

//a class declared with abstract modifier
//does not support activation
public abstract class Account
{
    public long Id { get; internal set; }

    public double Balance { get; protected set; }

    //a method declared with abstract modifier must
    //be overridden in a non-abstract derived class
    public abstract void Deposit(double amount);

    public abstract void Withdraw(double amount);

    public void Transfer(double amount, Account that)
    {
        if(!ReferenceEquals(this, that))
        {
            this.Withdraw(amount);
            that.Deposit(amount);
        }
    }
}